lib/services/upstart.c

```diff
@@ -365,30 +365,45 @@ upstart_job_metadata(const char *name)
     return crm_strdup_printf(METADATA_FORMAT, name, name, name);
 }
 
-static bool
-upstart_mask_error(svc_action_t *op, const char *error)
+/*!
+ * \internal
+ * \brief Set an action result based on a method error
+ *
+ * \param[in] op     Action to set result for
+ * \param[in] error  Method error
+ */
+static void
+set_result_from_method_error(svc_action_t *op, const DBusError *error)
 {
-    crm_trace("Could not issue %s for %s: %s", op->action, op->rsc, error);
-    if(strstr(error, UPSTART_06_API ".Error.UnknownInstance")) {
```

Comment:
why did gcc actually build this - does it really check the first element in the struct?

Reply:
the previous code took error->name as the argument instead of error